dts/arm/ti/am64x_m4.dtsi

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <freq.h>
+#include <arm/armv7-m.dtsi>
+#include <zephyr/dt-bindings/pinctrl/ti-k3-pinctrl.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+
+/ {
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4f";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@0 {
+		compatible = "mmio-sram";
+		reg = <0x0 DT_SIZE_K(192)>;		/* 192 KB of SRAM (I-Code) */
+	};
+
+	sram1: memory1@40000 {
+		compatible = "mmio-sram";
+		reg = <0x40000 DT_SIZE_K(64)>;		/* 64 KB of SRAM (D-Code) */
+	};
+
+	sysclk: system-clock {
+		compatible = "fixed-clock";
+		clock-frequency = <DT_FREQ_M(400)>;
+		#clock-cells = <0>;
+	};
+
+	pinctrl: pinctrl@4084000 {
+		compatible = "ti,k3-pinctrl";
+		reg = <0x04084000 0x88>;
+		status = "okay";
+	};
+
+	uart0: serial@4a00000 {
+		compatible = "ns16550";
+		reg = <0x04a00000 0x200>;
+		interrupts = <24 4>;
+		interrupt-parent = <&nvic>;
+		clock-frequency = <DT_FREQ_M(96)>;
+		current-speed = <115200>;
+		reg-shift = <2>;
+		status = "disabled";
+	};
+
+	uart1: serial@4a10000 {
+		compatible = "ns16550";
+		reg = <0x04a10000 0x200>;
+		interrupts = <25 4>;
+		interrupt-parent = <&nvic>;
+		clock-frequency = <DT_FREQ_M(96)>;
+		current-speed = <115200>;
+		reg-shift = <2>;
+		status = "disabled";
+	};
+
+	gpio0: gpio@4201010 {
+		compatible = "ti,davinci-gpio";
+		reg = <0x4201010 0x100>;
+		#address-cells = <2>;
```

Comment:
What does `#address-cells` do here?
I though `#gpio-cells` specified the number of cells for child bindings

Reply:
Yeah that looks wrong. If I recall correctly, I added that because the build failed due missing address cells in earlier version. Anyhow, gonna remove that from both m4 dtsi files since it's working now.